packages/test/src/workflows/signal-update-ordering.ts

```diff
@@ -0,0 +1,18 @@
+import * as wf from '@temporalio/workflow';
+
+export const fooSignal = wf.defineSignal('foo');
+export const fooUpdate = wf.defineUpdate<number>('foo');
+
+// Repro for https://github.com/temporalio/sdk-typescript/issues/1474
+export async function signalUpdateOrderingWorkflow(): Promise<number> {
+  let numFoos = 0;
+  wf.setHandler(fooSignal, () => {
+    numFoos++;
+  });
+  wf.setHandler(fooUpdate, () => {
+    numFoos++;
+    return numFoos;
+  });
```

Comment:
Also maybe slightly overkill, but, just to make sure somehow a handler didn't get double-run it might make sense to store `numFoos` as `totalHandlersRun` and then separately `signalsRun` and `updatesRun` and verify all three things

Reply:
I added a distinct, more comprehensive unit test that traces important events as perceived by the workflow. I'll this test as is.